api/handlers.go

```diff
@@ -0,0 +1,333 @@
+	if err != nil {
+		a.HTTPError(w, err)
+		return
+	}
+
+	if vol, err := a.Client.GetVolume(volume.Policy, volume.Name); err == nil && vol != nil {
+		a.HTTPError(w, errors.Exists)
+		return
+	}
+
+	log.Infof("Creating volume %s/%s", volume)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		a.HTTPError(w, errors.GetHostname.Combine(err))
+		return
+	}
+
+	policyObj, err := a.Client.GetPolicy(volume.Policy)
+	if err != nil {
+		a.HTTPError(w, errors.GetPolicy.Combine(errored.New(volume.Policy)).Combine(err))
+		return
+	}
+
+	uc := &config.UseMount{
+		Volume:   volume.String(),
+		Reason:   lock.ReasonCreate,
+		Hostname: hostname,
+	}
+
+	snapUC := &config.UseSnapshot{
+		Volume: volume.String(),
+		Reason: lock.ReasonCreate,
+	}
+
+	global := *a.Global
+
+	err = lock.NewDriver(a.Client).ExecuteWithMultiUseLock([]config.UseLocker{uc, snapUC}, global.Timeout, func(ld *lock.Driver, ucs []config.UseLocker) error {
+		volConfig, err := a.Client.CreateVolume(volume)
+		if err != nil {
+			return err
+		}
+
+		log.Debugf("Volume Create: %#v", *volConfig)
+
+		do, err := control.CreateVolume(policyObj, volConfig, global.Timeout)
+		if err == errors.NoActionTaken {
+			goto publish
+		}
+
+		if err != nil {
+			return errors.CreateVolume.Combine(err)
+		}
+
+		if err := control.FormatVolume(volConfig, do); err != nil {
+			if err := control.RemoveVolume(volConfig, global.Timeout); err != nil {
+				log.Errorf("Error during cleanup of failed format: %v", err)
+			}
+			return errors.FormatVolume.Combine(err)
+		}
+
+	publish:
+		if err := a.Client.PublishVolume(volConfig); err != nil && err != errors.Exists {
+			if _, ok := err.(*errored.Error); !ok {
+				return errors.PublishVolume.Combine(err)
+			}
+			return err
+		}
+
+		return a.WriteCreate(volConfig, w)
+	})
+
+	if err != nil && err != errors.Exists {
+		a.HTTPError(w, errors.CreateVolume.Combine(err))
+		return
+	}
+}
+
+func (a *API) get(origName string, r *http.Request) (string, error) {
+	policy, name, err := storage.SplitName(origName)
+	if err != nil {
+		return "", errors.GetVolume.Combine(err)
+	}
+
+	driver, volConfig, driverOpts, err := a.GetStorageParameters(&Volume{Policy: policy, Name: name})
+	if err != nil {
+		return "", errors.GetVolume.Combine(err)
+	}
+
+	if err := volConfig.Validate(); err != nil {
+		return "", errors.ConfiguringVolume.Combine(err)
+	}
+
+	path, err := driver.MountPath(driverOpts)
+	if err != nil {
+		return "", errors.MountPath.Combine(err)
+	}
+
+	return path, nil
+}
+
+func (a *API) writePathError(w http.ResponseWriter, err error) {
+	if err, ok := err.(*errored.Error); ok && err.Contains(errors.NotExists) {
+		w.Write([]byte("{}"))
+		return
+	}
+	a.HTTPError(w, err)
+	return
+}
+
+func (a *API) getMountPath(driver storage.MountDriver, driverOpts storage.DriverOptions) (string, error) {
+	path, err := driver.MountPath(driverOpts)
+	return path, err
+}
+
+// Path is the handler for both Path and Remove requests. We do not honor
```

Comment:
Can you explain what is meant by "we do not honor remove requests" here?  What happens if one is sent in?